### ISSUE TITLE
CS0104: 'Job' is an ambiguous reference between 'BenchmarkDotNet.Jobs.Job' and 'Nest.Job'

### ISSUE THREAD

**Issue by russcam** [dotnet/BenchmarkDotNet#778]
I'm looking to run benchmarks between two different versions of [NEST, the Elasticsearch .Net client](https://github.com/elastic/elasticsearch-net). Upon running a simple Program with `BenchmarkSwitcher`, a Build Exception is thrown

```
// ***** BenchmarkRunner: Start   *****
// Found benchmarks:
//   GithubIssue3236_Get562_SingleRun_X64_Jit-RyuJit_Runtime-Clr_WarmupCount1_TargetCount1_LaunchCount5
//   GithubIssue3236_Get610_SingleRun_X64_Jit-RyuJit_Runtime-Clr_WarmupCount1_TargetCount1_LaunchCount5

// Validating benchmarks:
// **************************
// Benchmark: GithubIssue3236_Get562_SingleRun_X64_Jit-RyuJit_Runtime-Clr_WarmupCount1_TargetCount1_LaunchCount5
// *** Generate *** 
// Result = Success
// BinariesDirectoryPath = C:\Users\russc\source\git\NestVersionBenchmark\NestVersionBenchmark\bin\Release

// *** Build ***
BuildScript: C:\Users\russc\source\git\NestVersionBenchmark\NestVersionBenchmark\bin\Release\BDN.Auto.bat
CS0104: 'Job' is an ambiguous reference between 'BenchmarkDotNet.Jobs.Job' and 'Nest562.Job'// Result = Failure
```

_For additional context, I've rewritten two different versions of NEST assembly and their immediate non-framework dependencies, to allow them both to be referenced in one benchmarking project. I think this is not relevant to the problem at hand, but mention it nonetheless_.

Since a Build Exception is thrown, benchmarking doesn't run. Any idea how I might go about fixing this issue? I'm guessing this is coming from the Roslyn Toolchain `Builder` compilation step?

**Comment by adamsitnik** [comment]
Hi @russcam 

To access the auto-generated code you need to apply a `[KeepBencharkFiles]` attribute on the type with benchmarks. (we remove all these files by default)

I can see that we have following line in the code `var job = new Job();`

Is `Nest562.Job` a type created for benchmarking purpose? If so, please rename it.

I will of course change this line on our side.

**Comment by russcam** [comment]
Thanks for the quick response @adamsitnik.

> Is Nest562.Job a type created for benchmarking purpose? If so, please rename it.

No, it's a type in the rewritten Nest assembly for version 5.6.2; it's not referenced/used in the benchmark test.

**Comment by adamsitnik** [comment]
We also generate `using` statements for all the types returned from the benchmarks. I guess that one of your benchmarks returns a type from `Nest562` namespace. So possible workaround would be to change the return type of the benchmark.

**Comment by russcam** [comment]
Awesome, thanks for the quick turnaround @adamsitnik :+1: Do you know when it'd be making its way to a released package?

**Comment by adamsitnik** [comment]
@russcam it's already available as part of `0.10.14.579` on our CI feed `https://ci.appveyor.com/nuget/benchmarkdotnet`

But release to nuget.org  (0.11.0) is +- 2 weeks

**Comment by russcam** [comment]
Perfect, I'll jump in the CI feed. Thanks!
